src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,83 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void register_addingNewUser_Ok() {
+        User userOk = new User();
+        userOk.setLogin("NewUser");
+        userOk.setPassword("123456789");
+        userOk.setAge(18);
+        User actual = registrationService.register(userOk);
+        assertEquals(userOk, actual);
+    }
+
+    @Test
+    void register_addingUserWhichAlreadyExist_NotOk() {
+        User userDouble = new User();
+        userDouble.setLogin("userDouble");
+        userDouble.setPassword("123456789");
+        userDouble.setAge(18);
+        registrationService.register(userDouble);
+        User actual = registrationService.register(userDouble);
+        assertNull(actual);
+    }
+
+    @Test
+    void register_addingUsersWithSameLogin_NotOk() {
+        User baseUser = new User();
+        baseUser.setLogin("twinUser");
+        baseUser.setPassword("123456789");
+        baseUser.setAge(18);
+        User twinUser = new User();
+        twinUser.setLogin("twinUser");
+        twinUser.setPassword("987654321");
+        twinUser.setAge(38);
+        registrationService.register(baseUser);
+        User actual = registrationService.register(twinUser);
+        assertNull(actual);

```

Comment:
Similarly, in the test `register_addingUsersWithSameLogin_NotOk`, expecting `null` for a duplicate login is not ideal. The service should handle this case by throwing an exception or providing a clear indication of the error. Consider revising the implementation and the test to align with this behavior.